src/routes/upload/index.tsx

```diff
@@ -0,0 +1,254 @@
+
+      if (indexWWW === -1 || indexWWW > indexLinkedIn) {
+        if (indexHTTPS !== -1) {
+          linkedIn =
+            linkedIn.substring(0, httpsLength) + 'www.' + linkedIn.substring(httpsLength)
+        } else {
+          linkedIn = 'www.' + linkedIn
+        }
+      }
+
+      if (indexHTTPS === -1) {
+        linkedIn = 'https://' + linkedIn
+      }
+    } else if (isValidLIUser(linkedIn)) {
+      linkedIn = `https://www.linkedin.com/in/${linkedIn}`
+    } else {
+      linkedIn = null
+    }
+    let major = event.target.major.value.trim()
+    if (this.state.majorOther) {
+      major = event.target.otherMajor.value.trim()
+    }
+
+    const info = {
+      name,
+      email,
+      linkedIn,
+      gpa,
+      major,
+      standing,
+    }
+
+    return info
+  }
+
+  handleSubmit = async (event: React.ChangeEvent<any>) => {
+    event.preventDefault()
+    event.stopPropagation()
+    event.persist()
+
+    const info = this.normalizeData(event)
+    if (event.target.pdf.files.length === 0) {
+      this.setState({ error: 'No file given' })
+      return false
+    }
+
+    if (!info.linkedIn) {
+      this.setState({ error: 'Not a valid LinkedIn URL' })
+      return false
+    }
+
+    //Convert pdf to base64 since Netlify/AWS Lambda doesn't allow binary content
+    let base64 = await this.toBase64(event.target.pdf.files[0])
+    let formData = new FormData()
+
+    formData.append('name', info.name)
+    formData.append('email', info.email)
+    formData.append('linkedin', info.linkedIn)
+    formData.append('gpa', info.gpa)
+    formData.append('major', info.major)
+    formData.append('standing', info.standing)
+    formData.append('pdf', base64 as string)
+
+    const endpoint_url = `${ENDPOINT}/api/file`
+    fetch(endpoint_url, {
+      method: 'POST',
+      body: formData,
+    }).then((res) => {
+      if (res.status !== 200) {
+        this.setState({
+          error: `While uploading, there was a ${res.status} error`,
+          success: false,
+        })
+      } else {
+        this.setState({ error: '', success: true })
+      }
+    })
+  }
+
+  render() {
+    const classes = this.props.classes
+    return (
+      <div className={classes.root}>
+        <NavBar enableDrawer={false} classes={classes} />
+        <div className={classes.content}>
+          <div className={classes.toolbar} />
+          <Grid container alignItems="center" direction="column" justify="center">
+            <p className="title">Upload Information</p>
+            <span className="sub-text">*Note: all fields are required</span>
+            {this.state.error !== '' ? (
+              <span className="sub-text red">{this.state.error}</span>
+            ) : null}
+            {this.state.success ? (
+              <span className="sub-text green">Your resume has been added</span>
+            ) : null}
+            <form
+              className="dg-form"
+              onSubmit={this.handleSubmit}
```

Comment:
```suggestion
              onSubmit={() => { this.setState({currentlySubmitting: true}); this.handleSubmit(); this.setState({currentlySubmitting: false})}
```

Reply:
just a note so you know, lamdas are more inefficient because they are reinitialized on every render